src/Controller/ApplePay.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Payplug\PayplugWoocommerce\Controller;
+
+use Payplug\Exception\HttpException;
+use Payplug\PayplugWoocommerce\Gateway\PayplugAddressData;
+use Payplug\PayplugWoocommerce\PayplugWoocommerceHelper;
+use Payplug\PayplugWoocommerce\Gateway\PayplugGateway;
+use Payplug\Resource\Payment as PaymentResource;
+use Payplug\PayplugWoocommerce\Gateway\PayplugGatewayOney3x;
+
+class ApplePay extends PayplugGateway
+{
+
+	protected $enable = false;
+	protected $domain_name = "";
+
+	public function __construct()
+	{
+
+		parent::__construct();
+
+		/** @var \WC_Settings_API  override $id */
+		$this->id = 'apple_pay';
+
+		/** @var \WC_Payment_Gateway overwrite for apple pay settings */
+		$this->method_title = __('payplug_apple_pay_title', 'payplug');
+		$this->method_description = __('payplug_apple_pay_description', 'payplug');
+
+		$this->title = 'payplug_apple_pay_title';
+		$this->description = '<div id="apple-pay-button-wrapper"><apple-pay-button buttonstyle="black" type="pay" locale="'. get_locale() .'"></apple-pay-button></div>';
+		$this->domain_name = strtr(get_site_url(), array("http://" => "", "http://" => ""));
+
+		if (!$this->checkApplePay()) {
+			$this->enabled = 'no';
+		} else {
+			add_action('wp_enqueue_scripts', [$this, 'add_apple_pay_css']);
+			add_action('wp_enqueue_scripts', [$this, 'add_apple_pay_js']);
+			add_action('woocommerce_before_checkout_form', [$this, 'woocommerce_before_checkout_form_callback']);
+		}
+
+	}
+
+	public function woocommerce_before_checkout_form_callback() {
+		echo '<apple-pay-button buttonstyle="black" type="pay" locale="'. get_locale() .'"></apple-pay-button>';
+	}
+
+	/**
+	 * Check Apple Pay Availability
+	 *
+	 */
+	private function checkApplePay(){
+		$account = PayplugWoocommerceHelper::get_account_data_from_options();
+
+		if (isset($account['payment_methods']['apple_pay']['enabled']) ) {
+
+			if( !empty($account['apple_pay']) && $account['apple_pay'] === 'yes' ) {
+				$applepay = false;
+				if ($account['payment_methods']['apple_pay']['enabled']) {
+					if (in_array(strtr(get_site_url(), array("http://" => "", "http://" => "")), $account['payment_methods']['apple_pay']['allowed_domain_names'])) {
```

Comment:
what about https?
why not use the same comparison we use in the other classes? (country_code_billing)

Reply:
Yeah I think it should be like this 
get_site_url(), array("http://" => "", "https://" => ""))

Reply:
what if the users uses a ".com" domain, it will not be able to use apple pay :) maybe we should use the billing address like we use on the other payment methods? this has been discussed?

Reply:
nevermind just understood :)
we only need to update for https I can do it :)